### ISSUE TITLE
(Windows only) Text in tab bar is sometimes unreadable depending on brave colors and windows accent color

### ISSUE THREAD

**Issue by dokugaron** [brave/brave-browser#22027]
## Description 
On windows, having a lighter accent color causes the text on unselected tabs, as well as the new tab icon in the tab bar to become dark and unreadable when using the default dark theme. Having a darker accent color causes the text to become unreadable in the light color theme

## Steps to Reproduce
   1. install `1.37.109` on windows
   2. set `accent color` in `windows settings > Personalization > Colors` to a lighter color such as green or red
   3. scroll down to `Show accent color on the following surfaces` and make sure that `Title bars and window borders` is checked
   4. launch brave
   5. go to `brave://settings/appearance`
   6. set `Brave colors` to `Dark`
   7. open more than one tab

## Actual result:
Tab text and new tab icon is black and pretty much unreadable.
![image](https://user-images.githubusercontent.com/82846166/161150556-8cb368d9-1afc-461a-a5d4-8cafe821c893.png)

## Expected result:
Tab text and new tab icon should be a lighter color in the dark theme

## Reproduces how often: 
Only on when `Brave colors` is `Dark` and accent color is a lighter color, or when `Brave colors` is `Light` and accent color is a darker color

## Brave version (brave://version info)

<html>
<body>

Brave | 1.37.109 Chromium: 100.0.4896.60 (Official Build) (64-bit)
-- | --
Revision | 6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875}
OS | Windows 10 Version 21H1 (Build 19043.1586)

</body>
</html>

## Version/Channel Information:
The problem is also present on the current nightly release `1.39.22`

## Other Additional Information:
The same behavior happens on chromium, but since the top bar on chromium is set to the color of the windows accent color, the text is still readable. Since the top bar is always dark colored in the brave dark theme, the text just becomes unreadable.

How it looks in chromium `102.0.4977.0` side by side with how it looks in brave `1.37.109`: 
![image](https://user-images.githubusercontent.com/82846166/161153832-bc6c19ed-1ff6-496c-818b-d889d925ab75.png)

With a darker color the text becomes white:
![image](https://user-images.githubusercontent.com/82846166/161153872-9c92107c-b8d2-4b9b-9b2f-7840b5016fb7.png)

However, it becomes unreadable in the light theme:
![image](https://user-images.githubusercontent.com/82846166/161153977-ca39d643-ea70-4983-9ce7-28488ca55fb4.png)

**Comment by aka13-404** [comment]
![image](https://user-images.githubusercontent.com/51682059/161215449-54d5efaa-8f50-463c-aa3a-5ab191ece6bb.png)
Can confirm, same thing happened to me as well.
Windows 10, 21H2 19044.1620, latest Brave version.

**Comment by Column01** [comment]
Same issue here:
![image](https://user-images.githubusercontent.com/26526649/161289118-6ac714ed-a2bd-478d-ae2e-860e8e45fcf7.png)

**Comment by mozomoid** [comment]
Hi there!
Yet another exemple:

![Capture d’écran 2022-04-02 021947](https://user-images.githubusercontent.com/37461926/161356490-8d3903ce-ac6a-4e20-b274-2b2a78426f47.jpg)

Win10 Pro 21H2 19044.1586
Brave 1.37.109

**Comment by Meatclump** [comment]
Same here.
Windows 10 Pro
19042.1526
Brave 1.37.109
![image](https://user-images.githubusercontent.com/11631528/161358120-b0445631-77d8-4a75-846a-df69715935df.png)

**Comment by francesco-ferraro** [comment]
Same issue here. Windows 10

![image](https://user-images.githubusercontent.com/39172206/161389093-32530c21-68e7-42eb-843a-112644f965ff.png)

**Comment by wbogocki** [comment]
Likewise, Windows 11.

**Comment by nhantrn** [comment]
Same.

Active window:
![2022-04-04_131906](https://user-images.githubusercontent.com/3202654/161625309-b5eeb4d8-c619-47d5-99b5-582421904be3.jpg)

Inactive window:
![2022-04-04_131916](https://user-images.githubusercontent.com/3202654/161625343-313d0d0b-9dff-4315-962c-4b94209d5330.jpg)

I think the problem is the new color scheme engine in Brave tries to match the color of Windows accent color setting, but forgot to also applies that to the main windows background color. Thus the contrast is out of whack.

What it should be like, Edge browser as an example:
![2022-04-04_132340](https://user-images.githubusercontent.com/3202654/161625916-e0307133-f575-4882-b035-17d94b4d7900.jpg)

**Comment by MadhaviSeelam** [comment]
Reproduced the issue following above test plan with dark and light colors

Brave | 1.37.109 Chromium: 100.0.4896.60 (Official Build) (64-bit)
-- | --
Revision | 6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875}
OS | Windows 11 Version 21H2 (Build 22000.593)

Steps:

<summary>Scenario 1: Brave colors: Dark</summary>
<details>

install 1.37.109 
set accent color in windows settings > Personalization > Colors to `Green `
scroll down to ` Show accent color on the title bars ad windows borders` - toggle on
launch brave
go to `brave://settings/appearance`
set Brave colors to `Dark`
open more than one tab

Actual Results: Inactive tab text is unreadable

<img width="950" alt="tab bar dark" src="https://user-images.githubusercontent.com/98358127/162023764-51c01883-fd56-45c9-9310-57ff196b70c8.png">

</details>

<summary>Scenario 2: Brave colors: Light</summary>
<details>

install 1.37.109 
set accent color in windows settings > Personalization > Colors to `Storm`
scroll down to  `Show accent color on the title bars ad windows borders` - toggle on
launch brave
go to brave://settings/appearance
set Brave colors to `Light`
open more than one tab

Actual Results: Actual Results: Inactive tab text is unreadable

<img width="950" alt="tab bar light" src="https://user-images.githubusercontent.com/98358127/162023934-9efe7afc-36e5-40bb-ae87-86828f533731.png">
</details>

**Comment by amzero** [comment]
Same here

![github](https://user-images.githubusercontent.com/6620339/162056841-cb7c6860-569a-4123-b774-14c15abba09b.png)

Windows 8.1
Brave Browser 1.37.111

**Comment by dalmo3** [comment]
For the record, enabling `brave://flags/#pervasive-system-accent-color` makes no difference.

**Comment by GeetaSarvadnya** [comment]
Reproduced the issue with Green/Red accent colors - `Windows 10 x64 - 1.37.111`

system settings screen 1 | system settings screen 2 | Brave tabs unreadable
-- | -- | --
![image](https://user-images.githubusercontent.com/38657976/162141547-6eae6869-a188-4370-9300-e4ca36878839.png) | ![image](https://user-images.githubusercontent.com/38657976/162141607-690d1766-5575-4f29-af47-f2082155b642.png) | ![image](https://user-images.githubusercontent.com/38657976/162141673-fe057d0b-ba4f-4c92-bd4b-44115c9aa424.png)

**Comment by kjozwiak** [comment]
Above requires `1.37.114` or higher for `1.37.x` verification.

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```
Brave | 1.37.114 Chromium: 100.0.4896.88 (Official Build) (64-bit)
-- | --
Revision | 4450653bfa91182e85723d8f1dee64dd6ce40ed4-refs/branch-heads/4896@{#1086}
OS | Windows 10 Version 21H2 (Build 19044.1586)

```

**_Dark theme_**


**_Light theme_**

**Comment by amzero** [comment]
Today my brave updated, but the bug was not fixed here, aparentely. I'm doing something wrong?

![brave](https://user-images.githubusercontent.com/6620339/163243082-b461c49c-247a-4c1b-ab5b-e46d1129669b.jpg)

Windows 8.1

**Comment by LaurenWags** [comment]
Hi @amzero - we're aiming to release a version with this fix tomorrow. Stay tuned!

**Comment by tomingle** [comment]
Still not working for me with today's update:

`Version 1.37.114 Chromium: 100.0.4896.88 (Official Build) (64-bit)`
`Windows 8.1 v6.3.9600`
`Settings > Appearance > Brave Colours: Light`

https://user-images.githubusercontent.com/6359614/163468504-88b11a39-abe4-43fd-8342-a47402aa6885.mp4

Normal windows are fine, inactive tab text in incognito windows is unreadable.

**Comment by amzero** [comment]
![brave](https://user-images.githubusercontent.com/6620339/163471297-60ce8669-6a06-4ec5-8ba7-dd21cfd05ee3.jpg)

Windows 8.1
Versão 1.37.114 Chromium: 100.0.4896.88 (Versão oficial) 64 bits

**Comment by gurnec** [comment]
The update *did* work for me.

Version 1.37.116 Chromium: 100.0.4896.127 (Official Build) (64-bit)
Windows 10 19044.1645

It did not work for me on a (not updated / old) Windows 8.1 VM I gave it a quick try on, it looked like the screenshots added by the two comments directly above. Trying high-contrast themes further changed the text color (for the worse).

**Comment by amzero** [comment]
I'm new in GitHub. The problem wasn't fixed on Windows 8.1. I'm not finding an opened issue to deal with it, 
Do we have to open a new one?

**Comment by kjozwiak** [comment]
> I'm new in GitHub. The problem wasn't fixed on Windows 8.1. I'm not finding an opened issue to deal with it, Do we have to open a new one?

Thanks for all the reports 👍 I managed to reproduce the issue and created https://github.com/brave/brave-browser/issues/22365 as a follow up for `Win 8`. As per the confirmations above, looks like the issue was resolved for both `Win 11` & `Win 10` but is still a problem within `Win 8`. Assuming it's due to `Win 8` using a different theming/colour implementation compared to `Win 11` & `Win 10`. We'll try getting this fixed ASAP 👍
